core/src/main/java/org/kiwix/kiwixmobile/core/search/viewmodel/SearchResultGenerator.kt

```diff
@@ -40,15 +40,24 @@ class ZimSearchResultGenerator @Inject constructor(
     if (sharedPreferenceUtil.prefFullTextSearch)
       zimReaderContainer.search(it, 200).run { fullTextResults() }
     else
-      zimReaderContainer.searchSuggestions(it, 200).run { suggestionResults() }
+      zimReaderContainer.searchSuggestions(it, 200).run { suggestionResults(it) }
 
   private fun fullTextResults() = generateSequence {
     zimReaderContainer.getNextResult()?.title?.let(::ZimSearchResultListItem)
   }.filter { it.value.isNotBlank() }
     .toList()
 
-  private fun suggestionResults() = generateSequence {
-    zimReaderContainer.getNextSuggestion()?.let { ZimSearchResultListItem(it.title) }
-  }.distinct()
-    .toList()
+  private fun suggestionResults(searchTerm: String) =
+    generateSequence {
+      zimReaderContainer.getNextSuggestion()?.let { ZimSearchResultListItem(it.title) }
+    }.distinct()
+      .toList()
+      .run {
+        if (none { it.value == searchTerm })
+          this
+        else
+          (sequenceOf(ZimSearchResultListItem(searchTerm)) + asSequence().filter {
```

Comment:
we don't need any sequence here

Reply:
I originally wanted to avoid a list copy. If the list copy is inevitable, then I might as well move it to `State.Results`

Reply:
It already has been copied, we called toList before this which created a list from a sequence, and at the end of this we call toList again. While it is a noble ideal to avoid excessive list creation it is usually unnecessary and the code reads more expressively without it like in my example code using `+` and `-` operators

Reply:
In `State.Results`, I used the `+` operator as you suggested. TIL.